clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -899,6 +899,10 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
   if (CodeGenOpts.PointerAuth.IndirectGotos)
     Fn->addFnAttr("ptrauth-indirect-gotos");
 
+  // Add return control flow integrity attributes.
+  if (CodeGenOpts.CFProtectionReturn)
+    Fn->addFnAttr("hw-shadow-stack");
+
```

Comment:
This feel a bit weird: I thought using `-fcf-protection=return` to select hardware-based shadow stack is a RISC-V specific thing. Perhaps for now this should look more tailored for RISC-V to avoid misleading?

Reply:
Yeah nice catch, it definitely needs to check the arch